src/Compilers/Core/MSBuildTask/CommandLineBuilderExtension.cs

```diff
@@ -110,6 +117,42 @@
             AppendTextWithQuoting(parameter);
         }
 
+        /// <summary>
+        /// Appends a switch with a value that is force quoted. This will quote as if <see cref="CommandLineBuilder.IsQuotingRequired(string)"/>
+        /// returns true for <paramref name="parameter"/>. That means even simple values will be quoted.
+        /// </summary>
+        internal void AppendTextWithForceQuoting(string parameter)
+        {
+            Debug.Assert(!_isQuotingRequired);
+            _isQuotingRequired = true;
+            try
+            {
+                AppendTextWithQuoting(parameter);
+            }
+            finally
+            {
+                _isQuotingRequired = false;
+            }
+        }
+
+        /// <summary>
+        /// Appends a switch with a value that is force quoted. This will quote as if <see cref="CommandLineBuilder.IsQuotingRequired(string)"/>
+        /// returns true for <paramref name="parameter"/>. That means even simple values will be quoted.
+        /// </summary>
+        internal void AppendSwitchForceQuoted(string switchName, string parameter)
```

Comment:
>AppendSwitchForceQuoted [](http://example.com/codeflow?start=22&length=23)

nit: is there a reason to use a different naming convention than method above (`Append...WithForceQuoting`)?

Reply:
The naming convention for these methods is roughly `Append(Switch|Text)?(QuoteStlye)?(IfNotNull)?`. This type is extending `CommandLineBuilder` where conventions like this are established. The current form is matching the style that I see there. 

Consider as an example `CommandLineBuilder.AppendSwitchUnquotedIfNotNull`.